src/Raven.Server/Documents/Indexes/Static/Counters/DynamicCounterEntry.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Diagnostics;
+using Raven.Server.Utils;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Indexes.Static.Counters
+{
+    public class DynamicCounterEntry : AbstractDynamicObject
+    {
+        private CounterGroupItemMetadata _counterItemMetadata;
+
+        private long _value;
+
+        public override dynamic GetId()
+        {
+            if (_counterItemMetadata == null)
+                return DynamicNullObject.Null;
+
+            Debug.Assert(_counterItemMetadata.Key != null, "_counterItemMetadata.Key != null");
+
+            return _counterItemMetadata.Key;
+        }
+
+        public override bool Set(object item)
+        {
+            _counterItemMetadata = (CounterGroupItemMetadata)item;
+            _value = 0;
+
+            var current = CurrentIndexingScope.Current;
+
+            var value = current.Index
+                .DocumentDatabase
+                .DocumentsStorage
+                .CountersStorage
+                .GetCounterValue(current.QueryContext.Documents, _counterItemMetadata.DocumentId, _counterItemMetadata.CounterName);
```

Comment:
`.QueryContext`? We call it during indexing not when querying it - please check it

Reply:
It is fine, in this case the name does not match very well, but indexing batch also needs to open that context, because we might be indexing compare exchange